content/en/docs/reference/command-line-tools-reference/feature-gates.md

```diff
@@ -441,6 +442,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `ScheduleDaemonSetPods`: Enable DaemonSet Pods to be scheduled by the default scheduler instead of the DaemonSet controller.
 - `SCTPSupport`: Enables the usage of SCTP as `protocol` value in `Service`, `Endpoint`, `NetworkPolicy` and `Pod` definitions
 - `ServerSideApply`: Enables the [Sever Side Apply (SSA)](/docs/reference/using-api/api-concepts/#server-side-apply) path at the API Server.
+- `ServiceAppProtocol`: Enables the `AppProtocol` field on Services and Endpoints.
```

Comment:
Does this work with EndpointSlices?

Reply:
EndpointSlices already have the field, so it does work with them, but it is not controlled by this feature gate.

Reply:
I agree with the case change suggested, but it feels strange to include EndpointSlices here. This feature gate has no direct effect on EndpointSlices. It allows Services to have `appProtocol` set, which would then get pulled into an EndpointSlice derived from that Service, but `appProtocol` can be set on an EndpointSlice regardless of the state of this feature gate.

Reply:
OK, that makes sense.
```suggestion
- `ServiceAppProtocol`: Enables the `appProtocol` field on Service and Endpoints objects.
```
(I haven't looked at the code for this at all).

Reply:
Updated, thanks!